Show countdown for clipboard clearing

ByteHamster opened the pull request:
[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Explain large or complex code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" )

Double-clicking to copy a password in the list is my most-used interaction with the application. Recently, I found myself clicking 3-4 times "just to be sure" because there is no visual indication that the password was actually copied. This PR now adds a progress bar and timer message to the status bar.

Closes #1843

## Screenshots
[TIP]:  # ( Do not include screenshots of your actual database! )

![image](https://user-images.githubusercontent.com/5811634/112232758-688c9400-8c39-11eb-9588-4d3f0a076d0b.png)

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and documented with doc blocks! )

Manual - clicked the item and watched the progress bar.

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ New feature (change that adds functionality)

droidmonkey commented:
I don't like this at all, sorry. Condense it to one side please.

ByteHamster commented:
Do you like this more?

![image](https://user-images.githubusercontent.com/5811634/112289973-137b6d00-8c8f-11eb-9639-12d93a9913fc.png)

I must admit that I'm pretty new to Qt. What do you think about the code structure?

droidmonkey commented:
The code is complicated but we can clean that up, there is no need for a signal slot to control the countdown.

ByteHamster commented:
> there is no need for a signal slot to control the countdown.

Could you give me a pointer what to do instead, please?

droidmonkey commented:
I am on the fence on whether the countdown updates should occur within MainWindow or the Clipboard class like you currently have. I want to re-use this progress bar setup for database decryption status as well. In order to make this more generic it would be good for the slot to accept a message along with the current progress.

Also, it appears that the status bar is unstyled and rather large, we probably need to add a style to it in the stylesheets.

droidmonkey commented:
I made some styling fixes for you. See what you can do to make this more generic for other uses.

ByteHamster commented:
I renamed the slot to something more generic and added a message parameter. Now multiple signals can be connected to the same slot. Was that what you had in mind?

droidmonkey commented:
Yes it is actually perfect! I van easily add messaging like "Transforming database key" and "Saving database". Thank you!

droidmonkey commented:
Fixed a bug where the progress bar would stay visible after database unlock.
